packages/x-charts/src/PieChart/PiePlot.tsx

```diff
@@ -118,15 +122,8 @@ function PiePlot(props: PiePlotProps) {
                     highlightScope={series[seriesId].highlightScope}
                     highlighted={highlighted}
                     faded={faded}
-                    onClick={
-                      onClick
-                        ? (event) =>
-                            onClick(
-                              item.id,
-                              item,
-                              event as React.MouseEvent<SVGPathElement, MouseEvent>,
-                            )
-                        : undefined
+                    onClick={(event) =>
+                      onClick?.(event, { type: 'pie', seriesId, dataIndex: index }, item)
```

Comment:
Just code simplification. Except if you see a drawback

Reply:
The reason for the check is the cursor pointer, the check is currently if the onClick is undefined, with your change all pie slices have cursor pointer.